### ISSUE TITLE
InfoLOOB loss

### ISSUE THREAD

**Issue by raunak-agarwal** [UKPLab/sentence-transformers#1337]
Hi,
Do you think InfoLOOB (formulation available [here](https://ml-jku.github.io/cloob/)) would be a good addition to this library? Seems like it outperforms InfoNCE in an image-text setting, might be worth experimenting on text pairs

**Comment by nreimers** [comment]
Looks interesting. Do you know if there is a Pytorch implementation?

**Comment by raunak-agarwal** [comment]
I think I already linked the implementation above. [This](https://github.com/ml-jku/cloob/blob/master/src/training/methods.py) one is the official implementation. 
And I also found it integrated [here](https://github.com/EleutherAI/magiCARP/blob/main/carp/pytorch/model/architectures/carp_cloob.py), although their API is slightly different than yours

**Comment by raunak-agarwal** [comment]
Is there any interest in implementing this?

**Comment by nreimers** [comment]
Yes, I think it would be quite interesting. Sadly I'm currently quite limited in terms of available time. So a PR and some tests would be really appreciated.
